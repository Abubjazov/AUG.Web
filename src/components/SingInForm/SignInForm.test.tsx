import { fireEvent, render, screen } from '@testing-library/react'
import MockedProvider from 'mockData/mockedReduxProvider'

import SignInForm from './SignInForm'

describe('SignInForm', () => {
  test('should render SignInForm default', () => {
    const mockFn = vi.fn()

    const { asFragment } = render(
      <MockedProvider>
        <SignInForm userFunction={mockFn} onSignIn={mockFn} />
      </MockedProvider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should call SignIn function', () => {
    const mockFn = vi.fn()

    render(
      <MockedProvider>
        <SignInForm userFunction={mockFn} onSignIn={mockFn} />
      </MockedProvider>,
    )

    fireEvent.click(screen.getByText('Sign in'))

    expect(mockFn).toHaveBeenCalledTimes(0)

    fireEvent.change(screen.getByPlaceholderText('email@example.com'), {
      target: { value: 'email@test.tst' },
    })

    fireEvent.change(screen.getByPlaceholderText('password'), {
      target: { value: '12345678' },
    })

    fireEvent.click(screen.getByText('Sign in'))

    expect(mockFn).toHaveBeenCalledTimes(1)
  })
})
