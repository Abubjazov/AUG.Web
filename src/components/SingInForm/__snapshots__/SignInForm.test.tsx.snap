// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignInForm > should render SignInForm default 1`] = `
<DocumentFragment>
  <div
    class="_root_e1c370"
  >
    <span
      class="_title_e1c370"
    >
      Sign 
      <span
        class="_red-text_e1c370"
      >
        in
      </span>
    </span>
    <div
      class="_root_672eb8"
    >
      <input
        data-testid="email-input"
        name="email"
        placeholder="email@example.com"
        type="email"
        value=""
      />
    </div>
    <div
      class="_root_672eb8"
    >
      <input
        data-testid="password-input"
        name="password"
        placeholder="password"
        type="password"
        value=""
      />
    </div>
    <div
      class="_buttons_e1c370"
    >
      <button
        class="_root_91d6ec _contained-red_91d6ec  _button_e1c370"
        data-testid="base-button"
        type="button"
      >
        Sign in
      </button>
      <button
        class="_root_91d6ec _outlined-white_91d6ec  _button_e1c370"
        data-testid="base-button"
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
</DocumentFragment>
`;
